drop-transfer/src/ws/server/v4.rs

```diff
@@ -616,11 +616,8 @@ impl handler::Downloader for Downloader {
     }
 
     async fn validate(&mut self, path: &Hidden<PathBuf>) -> crate::Result<()> {
-        let csum = tokio::task::block_in_place(|| {
-            let file = std::fs::File::open(&path.0)?;
-            let csum = file::checksum(&mut io::BufReader::new(file))?;
-            crate::Result::Ok(csum)
-        })?;
+        let file = std::fs::File::open(&path.0)?;
+        let csum = file::checksum(&mut io::BufReader::new(file)).await?;
```

Comment:
Just an idea, I am not familiar with tokio that well - is it possible to drive the task manually without writing our own executor? This would be nice to check that the `checksum` is actually behaving properly.

It's not necessary, just a thought

Reply:
I don't know what you mean exactly but you can call https://doc.rust-lang.org/std/future/trait.Future.html#tymethod.poll on the future

Reply:
Looks exactly what I wanted :) I was just curious if we can test that `checksum()` requires several invocations in order to complete. Because we know that the internal buffer is 8K if we would pass in a 20K file it should take exactly three invocations.

That would be nice to have a unit test for

Reply:
Done. I must admit it was quite a fun